ensime_shared/client.py

```diff
@@ -224,26 +202,10 @@ def tell_module_missing(self, name):
         msg = feedback["module_missing"]
         self.editor.raw_message(msg.format(name, name))
 
-    # TODO: this should be the Ensime class's responsibility, not EnsimeClient;
-    #       move (and eliminate self.vim) after fixing, see #294
-    def disable_plugin(self):
-        """Disable plugin temporarily, including also related plugins."""
-        self.log.debug('disable_plugin: in')
-
-        def threadsafe_vim(command):
-            """Threadsafe call if neovim, normal if vim."""
-            def normal_vim(e):
-                self.vim.command(command)
-            with catch(Exception, normal_vim):
-                self.vim.session.threadsafe_call(command)
```

Comment:
Good catch, this may be related to this error I've seen several times now:
```
error caught while executing async callback:
TypeError("'str' object is not callable",)
TypeError: 'str' object is not callable
 
the call was requested at
  File "/home/jvican/.nvim/plugged/ensime-vim/ensime_shared/util.py", line 51, in catch
    handler(str(e))
  File "/home/jvican/.nvim/plugged/ensime-vim/ensime_shared/client.py", line 171, in logger_and_close
    self.disable_plugin()
  File "/home/jvican/.nvim/plugged/ensime-vim/ensime_shared/client.py", line 236, in disable_plugin
    threadsafe_vim('set runtimepath-={}'.format(path))
  File "/home/jvican/.nvim/plugged/ensime-vim/ensime_shared/client.py", line 232, in threadsafe_vim
    self.vim.session.threadsafe_call(command)
```